lib/spack/spack/repo.py

```diff
@@ -1398,12 +1395,17 @@ def get_pkg_class(self, pkg_name: str) -> Type["spack.package_base.PackageBase"]
         class_name = nm.pkg_name_to_class_name(pkg_name)
 
         try:
+            if self.python_path:
+                sys.path.insert(0, self.python_path)
             module = importlib.import_module(fullname)
         except ImportError as e:
             raise UnknownPackageError(fullname) from e
         except Exception as e:
             msg = f"cannot load package '{pkg_name}' from the '{self.namespace}' repository: {e}"
             raise RepoError(msg) from e
+        finally:
+            if self.python_path:
+                sys.path.pop(0)
```

Comment:
```suggestion
                sys.path.remove(self.python_path)
```

I think this should remove the same thing it added. You're not guaranteed that `importlib.import_module` doesn't end up prepending something to `sys.path` due to someone's weird package. Yes that would be bad, maybe even horrible, but I think that is someone else's concern, and this function should just do no harm if it happens 🙈.

Reply:
good point